pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -541,44 +545,51 @@ func (b *BridgeBindMechanism) decorateConfig() error {
 	return nil
 }
 
-func (b *BridgeBindMechanism) loadCachedInterface(pid, name string) (bool, error) {
-	ifaceConfig, err := b.storeFactory.CacheForPID(pid).Read(name)
+func getPIDString(pid *int) string {
+	if pid != nil {
+		return fmt.Sprintf("%d", *pid)
+	}
+	return "self"
+}
+
+func (b *BridgeBindMechanism) loadCachedInterface() error {
+	ifaceConfig, err := b.storeFactory.CacheForPID(getPIDString(b.launcherPID)).Read(b.iface.Name)
 	if os.IsNotExist(err) {
-		return false, nil
+		return nil
 	}
 
 	if err != nil {
-		return false, err
+		return err
 	}
 
 	b.virtIface = ifaceConfig
-	return true, nil
+	return nil
 }
 
-func (b *BridgeBindMechanism) setCachedInterface(pid, name string) error {
-	err := b.storeFactory.CacheForPID(pid).Write(name, b.virtIface)
-	return err
+func (b *BridgeBindMechanism) setCachedInterface() error {
+	pid := *b.launcherPID
+	return b.storeFactory.CacheForPID(fmt.Sprintf("%d", pid)).Write(b.iface.Name, b.virtIface)
 }
 
-func (b *BridgeBindMechanism) loadCachedVIF(pid, name string) (bool, error) {
-	buf, err := ioutil.ReadFile(getVifFilePath(pid, name))
+func (b *BridgeBindMechanism) hasCachedInterface() bool {
+	return b.virtIface != nil
+}
+
+func (b *BridgeBindMechanism) loadCachedVIF(pid string) error {
+	buf, err := ioutil.ReadFile(getVifFilePath(pid, b.iface.Name))
 	if err != nil {
-		return false, err
+		return err
 	}
 	err = json.Unmarshal(buf, &b.vif)
 	if err != nil {
-		return false, err
+		return err
 	}
 	b.vif.Gateway = b.vif.Gateway.To4()
-	return true, nil
+	return nil
 }
 
-func (b *BridgeBindMechanism) setCachedVIF(pid, name string) error {
-	buf, err := json.MarshalIndent(&b.vif, "", "  ")
-	if err != nil {
-		return fmt.Errorf("error marshaling vif object: %v", err)
-	}
-	return writeVifFile(buf, pid, name)
+func (b *BridgeBindMechanism) setCachedVIF(pid string) error {
+	return setCachedVIF(*b.vif, pid, b.iface.Name)
```

Comment:
We have the launcherID as part of the BidgeBindMechanism we don't need it as argument

Reply:
Unit tests won't fly like that, we need to extend the existing interface caching scheme to also cache the VIFs.

It is explained in the commit msg of https://github.com/kubevirt/kubevirt/pull/5059/commits/f9df27437db90a4478ef1549c4f933e8e536a698:

"""
The PID must still be passed as an argument, to allow unit tests,
since the production code caches the VIF in one path, and reads the
cache from another (symlinked) path. On unit tests, this symlink
does not exist, and is overkill.
"""

Reply:
Then can we rename this as

```suggestion
func (b *BridgeBindMechanism) setCachedVIF(pidFilePath string) error {
	return setCachedVIF(*b.vif, pid, b.iface.Name)
```

Reply:
No. That is not a path to anything. It is a PID, in string format.

It will eventually be used (within setCachedVIF or friends) to compose the path you're mentioning.